zebrad/tests/acceptance.rs

```diff
@@ -1473,3 +1487,387 @@ where
+    };
+
+    Ok(block.transactions.to_vec())
+}
+
+/// Starts a state service using the provided `cache_dir` as the directory with the chain state.
+async fn start_state_service(
+    network: Network,
+    cache_dir: impl Into<PathBuf>,
+) -> Result<(
+    BoxStateService,
+    impl Service<
+        zebra_state::ReadRequest,
+        Response = zebra_state::ReadResponse,
+        Error = zebra_state::BoxError,
+    >,
+    LatestChainTip,
+    ChainTipChange,
+)> {
+    let config = zebra_state::Config {
+        cache_dir: cache_dir.into(),
+        ..zebra_state::Config::default()
+    };
+
+    remove_file_if_it_exists(config.db_path(network).join("LOCK")).await?;
+
+    Ok(zebra_state::init(config, network))
+}
+
+/// Spawns a zebrad instance to interact with lightwalletd, but without an internet connection.
+///
+/// This prevents it from downloading blocks. Instead, the `zebra_directory` parameter allows
+/// providing an initial state to the zebrad instance.
+fn spawn_zebrad_for_rpc_without_initial_peers(
+    zebra_directory: TempDir,
+) -> Result<(TestChild<TempDir>, SocketAddr)> {
+    let mut config = random_known_rpc_port_config()
+        .expect("Failed to create a config file with a known RPC listener port");
+
+    config.state.ephemeral = false;
+    config.network.initial_mainnet_peers = HashSet::new();
+    config.network.initial_testnet_peers = HashSet::new();
+
+    let mut zebrad = zebra_directory
+        .with_config(&mut config)?
+        .spawn_child(&["start"])?
+        .with_timeout(Duration::from_secs(60 * 60));
+
+    let rpc_address = config.rpc.listen_addr.unwrap();
+
+    zebrad.expect_stdout_line_matches(&format!("Opened RPC endpoint at {}", rpc_address))?;
+    zebrad.expect_stdout_line_matches("activating mempool")?;
+
+    Ok((zebrad, rpc_address))
+}
+
+/// Start a lightwalletd instance with its RPC server functionality enabled.
+///
+/// Returns the lightwalletd instance and the port number that it is listening for RPC connections.
+fn spawn_lightwalletd_with_rpc_server(
+    zebrad_rpc_address: SocketAddr,
+) -> Result<(TestChild<TempDir>, u16)> {
+    let lightwalletd_dir = testdir()?.with_lightwalletd_config(zebrad_rpc_address)?;
+
+    let lightwalletd_rpc_port = random_known_port();
+    let lightwalletd_rpc_address = format!("127.0.0.1:{lightwalletd_rpc_port}");
+
+    let mut arguments = vec!["--grpc-bind-addr".to_owned(), lightwalletd_rpc_address];
+
+    if let Ok(data_dir) = env::var("LIGHTWALLETD_DATA_DIR") {
+        arguments.push("--data-dir".to_owned());
+        arguments.push(data_dir);
+    }
+
+    let borrowed_arguments: Vec<&str> = arguments.iter().map(|string| string.as_ref()).collect();
+    let mut lightwalletd = lightwalletd_dir.spawn_lightwalletd_child(&borrowed_arguments)?;
+
+    lightwalletd.expect_stdout_line_matches("Starting gRPC server")?;
+    lightwalletd.expect_stdout_line_matches("Waiting for block")?;
```

Comment:
Question: Is there a `sendrawtransaction` log we can check for here?

Reply:
I couldn't find anything to match on :(

Reply:
No worries, that's fine, we can only do what we can do!